packages/@atjson/offset-core-components/src/components/color-editor.ts

```diff
@@ -0,0 +1,71 @@
+import WebComponent from '../mixins/component';
+
+export default class ColorEditor extends WebComponent {
+
+  static template = `<label>Color</label> <input type="color" name="text-color" class="text-color" /> <label>Background</label> <input type="color" name="background-color" class="background-color" /> &nbsp; <button class="cancel">❌</button><button class="save">✔️</button>&nbsp;`;
+
+  static events = {
+    'beforeinput': 'beforeInput',
+    'click .cancel': 'cursorBlur',
+    'click .save': 'onSave',
+    'keypress .text-color': 'handleKeypress'
+    'keypress .background-color': 'handleKeypress'
+  };
+
+  static observedAttributes = ['text-color', 'background-color'];
+
+  static style = `
+    :host {
+      display: flex;
+    }
+  `;
+
+  beforeInput(evt: Event) {
+    evt.stopPropagation();
+  }
+
+  handleKeypress(evt: KeyboardEvent) {
+    if (evt.keyCode === 13) {
+      this.onSave(evt);
+    }
+  }
+
+  onSave(evt: Event) {
+    let textColor = this.shadowRoot.querySelector('.text-color');
```

Comment:
@blaine Should this be grabbing the corresponding attribute instead of querying for the element?

`this.getAttribute('text-color')`

Reply:
The color picker is pretty primitive, so it uses `<input type=color>` and this is just getting the element to propagate it up, using `element.value` to fetch the actual color value.